platform/srf06-cc26xx/common/i2c.c

```diff
@@ -293,38 +300,82 @@ board_i2c_write_read(uint8_t *wdata, uint8_t wlen, uint8_t *rdata, uint8_t rlen)
+bool
+i2c_write_single_read_multi(uint8_t wdata, uint8_t *buf, uint8_t len)
+{
+  return i2c_write_read(&wdata, 1, buf, len);
+}
+/*---------------------------------------------------------------------------*/
 void
-board_i2c_select(uint8_t new_interface, uint8_t address)
+i2c_select(uint32_t new_pin_sda,
+                 uint32_t new_pin_scl,
+                 uint8_t new_slave_address,
+                 bool new_speed,
+                 uint32_t new_pin_pull)
 {
-  slave_addr = address;
+  PRINTF("I2C: Selecting: sda: %lu, scl: %lu, addr: %x, speed: %d,pull: %lu\n",
+         new_pin_sda, new_pin_scl, new_slave_address, new_speed, new_pin_pull);
 
   if(accessible() == false) {
-    board_i2c_wakeup();
+    i2c_wakeup();
   }
 
-  if(new_interface != interface) {
-    interface = new_interface;
+  /* Configure for new values only */
+  if(new_pin_sda != selected_pin_sda ||
+      new_pin_scl != selected_pin_scl ||
+      new_speed != selected_speed) {
+
+    /* If already selected, deselect first */
+    if(selected_pin_sda != I2C_PIN_NOT_SET ||
+      selected_pin_sda != I2C_PIN_NOT_SET) {
+      i2c_deselect();
+    }
 
     ti_lib_i2c_master_disable(I2C0_BASE);
 
-    if(interface == BOARD_I2C_INTERFACE_0) {
-      ti_lib_ioc_io_port_pull_set(BOARD_IOID_SDA, IOC_NO_IOPULL);
-      ti_lib_ioc_io_port_pull_set(BOARD_IOID_SCL, IOC_NO_IOPULL);
-      ti_lib_ioc_pin_type_i2c(I2C0_BASE, BOARD_IOID_SDA, BOARD_IOID_SCL);
-      ti_lib_ioc_pin_type_gpio_input(BOARD_IOID_SDA_HP);
-      ti_lib_ioc_pin_type_gpio_input(BOARD_IOID_SCL_HP);
-    } else if(interface == BOARD_I2C_INTERFACE_1) {
-      ti_lib_ioc_io_port_pull_set(BOARD_IOID_SDA_HP, IOC_NO_IOPULL);
-      ti_lib_ioc_io_port_pull_set(BOARD_IOID_SCL_HP, IOC_NO_IOPULL);
-      ti_lib_ioc_pin_type_i2c(I2C0_BASE, BOARD_IOID_SDA_HP, BOARD_IOID_SCL_HP);
-      ti_lib_ioc_pin_type_gpio_input(BOARD_IOID_SDA);
-      ti_lib_ioc_pin_type_gpio_input(BOARD_IOID_SCL);
-    }
+    ti_lib_ioc_io_port_pull_set(new_pin_sda, IOC_NO_IOPULL);
+    ti_lib_ioc_io_port_pull_set(new_pin_scl, IOC_NO_IOPULL);
+    ti_lib_ioc_pin_type_i2c(I2C0_BASE, new_pin_sda, new_pin_scl);
 
     /* Enable and initialize the I2C master module */
     ti_lib_i2c_master_init_exp_clk(I2C0_BASE, ti_lib_sys_ctrl_clock_get(),
-                                   true);
+                                   new_speed);
   }
+
+  selected_pin_sda = new_pin_sda;
+  selected_pin_scl = new_pin_scl;
+  selected_speed = new_speed;
+  selected_slave_addr = new_slave_address;
+  selected_pin_pull = new_pin_pull;
+}
+/*---------------------------------------------------------------------------*/
+void
+i2c_deselect()
+{
+  PRINTF("I2C: Deselecting sda: %lu, scl: %lu, addr: %x\n",
+         selected_pin_sda, selected_pin_scl, selected_slave_addr);
+
+  /* Deselect only if selected */
+  if(selected_pin_sda != I2C_PIN_NOT_SET &&
+      selected_pin_scl != I2C_PIN_NOT_SET) {
```

Comment:
Most of the time you are checking whether at least one of these pins is not set. Why did you choose to change logic here?